src/umode.rs

```diff
@@ -46,9 +51,175 @@ struct UmodeCpuArchState {
     trap_csrs: TrapRegs,
 }
 
+impl UmodeCpuArchState {
+    fn print(&self) {
```

Comment:
Instead of defining your own `print()` function, how about implementing `Display` or `Debug` instead?

Reply:
Good point.

Reply:
Done.